internal/mode/static/handler.go

```diff
@@ -194,6 +173,11 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, logger logr.Log
 	case state.EndpointsOnlyChange:
 		h.version++
 		cfg := dataplane.BuildConfiguration(ctx, gr, h.cfg.serviceResolver, h.version)
+		if depCtx, err := h.setDeploymentCtx(ctx, logger); err != nil {
+			logger.Error(err, "error setting deployment context for usage reporting")
+		} else {
```

Comment:
Ah, the reason I had this is because I used the inline `if depCtx, err` which confines the depCtx to the scope of the conditional. I can change this format if preferred.

Reply:
I see, that makes sense. You know I hate else statements lol, but I'll leave it up to you.